### PR TITLE
Bluetooth: controller: openisa: Fix sanitycheck failures 

### LINKED ISSUE #29967: sanitycheck fails to build samples/bluetooth/peripheral_hr/sample.bluetooth.peripheral_hr_rv32m1_vega_ri5cy
`./scripts/sanitycheck -s samples/bluetooth/peripheral_hr/sample.bluetooth.peripheral_hr_rv32m1_vega_ri5cy -l`

Get:

```
/home/galak/git/zephyr/subsys/bluetooth/controller/ll_sw/ull.c:248:7: error: 'BT_CTLR_ADV_SET' undeclared here (not in a function)
  248 |       BT_CTLR_ADV_SET + BT_CTLR_SCAN_SYNC_SET))
      |       ^~~~~~~~~~~~~~~
/home/galak/git/zephyr/subsys/bluetooth/controller/ll_sw/ull.c:252:15: note: in expansion of macro 'PDU_RX_POOL_SIZE'
  252 |  uint8_t pool[PDU_RX_POOL_SIZE];
      |               ^~~~~~~~~~~~~~~~
gmake[2]: *** [zephyr/subsys/bluetooth/controller/CMakeFiles/subsys__bluetooth__controller.dir/build.make:213: zephyr/subsys/bluet
```

### PR DESCRIPTION
**Pull request by cvinayak**
Fix sanitycheck failures when building
samples/bluetooth/peripheral_hr for rv32m1_vega_ri5cy
platform.

Updates from nRF platform related to 255 byte AD data
support have been directly copied into openisa port to
fix build failures.

Relates to commit c2fc629 ("Bluetooth: controller:
255 byte AD payload support").

Fixes #29967.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

### CODE REVIEW

**Review by George-Stefan — APPROVED** [review]

**Comment by cvinayak** [comment]
@galak do let me know a minimal commandline script I should run on my PR, as these regressions are happening too often on my PRs which pass PR CI.

With regards to this PR, you may merge once required approvals are logged.

**Review by carlescufi — APPROVED** [review]

**Pull request merged**
